source/Octopus.Tentacle.Client/Retries/RpcCallNoRetriesHandler.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Octopus.Tentacle.Client.Retries
+{
+    public class RpcCallNoRetriesHandler
+    {
+        public async Task ExecuteWithNoRetries(
+            Func<CancellationToken, Task> action,
+            bool abandonActionOnCancellation,
+            TimeSpan abandonAfter,
+            CancellationToken cancellationToken)
+        {
+            await ((Func<CancellationToken, Task>)(async ct =>
+            {
+                if (!abandonActionOnCancellation)
+                {
+                    await action(ct).ConfigureAwait(false);
+                }
+
+                using var abandonCancellationTokenSource = new CancellationTokenSource();
+                using (ct.Register(() =>
+                       {
+                           // Give the actionTask some time to cancel on it's own.
+                           // If it doesn't assume it does not co-operate with cancellationTokens and walk away.
+                           abandonCancellationTokenSource.TryCancelAfter(abandonAfter);
+                       }))
+                {
+                    var abandonTask = abandonCancellationTokenSource.Token.AsTask();
+
+                    try
+                    {
+                        var actionTask = action(ct);
```

Comment:
This will likely need to have [the fix](https://github.com/OctopusDeploy/OctopusTentacle/pull/540) Nathan did over the weekend applied to it.

Reply:
Yeah good catch, I've updated 👍🏻